src/Nethermind/Nethermind.Trie/Pruning/NoPruning.cs

```diff
@@ -22,9 +22,11 @@ public class NoPruning : IPruningStrategy
 
         public static NoPruning Instance { get; } = new NoPruning();
 
+        public bool ShouldCache => false;
```

Comment:
I think this should not be responsibility of the pruning strategy to decide whether we should cache or not

Reply:
I suggest renaming it to PruningEnabled because it is strictly connected with pruning. In this way, I think we will have an elegant interface.